pkg/sqlstore/impl/user/rowstojson.go

```diff
@@ -40,7 +40,7 @@ func getRowsData(rows *sql.Rows, numColumns int) ([][]interface{}, error) {
 	for rows.Next() {
 		scanArgs := make([]interface{}, numColumns)
 		for i := range scanArgs {
-			scanArgs[i] = new(interface{})
+			scanArgs[i] = &sqlstore.UserValue{}
```

Comment:
Always scan into `UserValue`

Reply:
I think this is a good idea and avoids having to refactor tests, potentially miss some other required changes being hidden by mocked test data. Will try this out.

Reply:
Cool, remember to include the extra copy logic that you found from the `Scanner` interface below to handle `[]byte` since that's the correct thing since we retain `[]bytes` append in L48. So prob mostly "moving" the logic that already exists into a `switch` between lines L47 and L48 iterating in `scanArgs...` or something like that.

Reply:
Hey @jsign I'm spinning my wheels a bit trying to figure out the type of the value pointed to by a `*interface{}`. I'm tempted to use reflection api, but is there any way to avoid that?

Reply:
You can do the same as I mentioned in the other comment [here](https://github.com/tablelandnetwork/go-tableland/blob/89637eb0cd23fe1fa1ca12a193ac37f94c3102d3/internal/router/controllers/user.go#L146)?

So prob: `switch (*xxx).(*interface{})).(type) {` ?

Reply:
oh yea, thanks. all this indirecting and `interface{}` stuff can get confusing.